langlib/lang.query/src/main/ballerina/src/lang.query/helpers.bal

```diff
@@ -141,6 +156,169 @@ public function consumeStream(stream<Type, error?> strm) returns error? {
     }
 }
 
+public type StreamOrderBy object {
+    public string[] sortFields;
+    public boolean[] sortTypes;
+
+    public function init(string[] sortFields, boolean[] sortTypes) {
+        self.sortFields = sortFields;
+        self.sortTypes = sortTypes;
+    }
+
+    public function topDownMergeSort(@tainted Type[] events) returns @tainted Type[]{
+        int index = 0;
+        int n = events.length();
+        Type[] b = [];
+        while (index < n) {
+            b[index] = events[index];
+            index += 1;
+        }
+        self.topDownSplitMerge(b, 0, n, events);
+        return events;
+    }
+
+    function topDownSplitMerge(@tainted Type[] b, int iBegin, int iEnd, @tainted Type[] a) {
+        if (iEnd - iBegin < 2) {
+            return;
+        }
+        int iMiddle = (iEnd + iBegin) / 2;
+        self.topDownSplitMerge(a, iBegin, iMiddle, b);
+        self.topDownSplitMerge(a, iMiddle, iEnd, b);
+        self.topDownMerge(b, iBegin, iMiddle, iEnd, a);
+    }
+
+    function topDownMerge(@tainted Type[] a, int iBegin, int iMiddle, int iEnd, @tainted Type[] b) {
+        int i = iBegin;
+        int j = iMiddle;
+
```

Comment:
unnecessary line break

Reply:
This function is no longer available since order by behavior was modified to use the array:sort() function